### DISCUSSION TITLE
Can eunomia-bpf enable the JIT or AOT of WAMR?

### DISCUSSION THREAD

**Discussion by ApsarasX** [eunomia-bpf/eunomia-bpf#43] [Q&A] [answered]
`eunomia-bpf` currently only uses WAMR's interpreter, but it is too slow to execute the wasm program using the interpreter.

So I want to enable WAMR's JIT in `eunomia-bpf`. Is this possible?

**Comment by yunwei37** [comment, marked as answer]
https://github.com/eunomia-bpf/eunomia-bpf/blob/468991d302e436abf51298f09764c6ce3d9050c5/ewasm/CMakeLists.txt#L184

可以参考这里，这里是 wamr 的编译选项，可以调整一下?

**Reply by ApsarasX** [reply]
嗯嗯，我知道是使用`WAMR_BUILD_JIT`来开启JIT，主要问题是JIT使用了mmap，eBPF里面支持使用mmap吗？

**Reply by yunwei37** [reply]
目前本质上还是用户态程序和内核态程序分开开发和分开运行的，只是使用 wasm 开发用户态程序，最后把编译好的包含内核态 ebpf 字节码的 JSON 和用户态的 wasm 打包成一个 wasm 模块进行分发。

在目标机器上 ebpf 虚拟机的在 ebpf 虚拟机里面跑，wasm

**Reply by yunwei37** [reply]
wasm 字节码在 wasm 虚拟机里面跑，所以应该不会有什么影响? wamr 这里也只是去调用了底下 eunomia-bpf 库 export 出来的几个函数进行加载，下面那个库应该也不会被 mmap 影响?
